lib/actions/fcs.py

```diff
@@ -339,6 +366,422 @@ def fcs2html(self, req):
+                    attributes = {}
+                    key, value = elem
+                else:
+                    key, value, attributes = elem
+
+                if type(value) == list:
+                    el = self._get_element_tree(ns, key, value, attributes)
+                    root.insert(MAX_INT, el)
+
+                else:
+                    el = ET.SubElement(root, ns + key)
+                    el.text = value
+                    el.attrib = attributes
+
+        return root
+
+    def _get_endpoint_description_tree(self, parameters, ns='ed'):
+        """
+        Creates xml.etree.ElementTree object for fcs endpoint description site.
+        :param parameters: dict
+        :param ns: str, namespace
+        :return: xml.etree.ElementTree
+        """
+        default_corpus_info = self._corpus_info(parameters['corpname'])
+
+        data = [
+            ('Capabilities', [('Capability', 'http://clarin.eu/fcs/capability/basic-search')]),
+            ('SupportedDataViews', [
+                ('SupportedDataView', 'application/x-clarin-fcs-hits+xml', {
+                    'id': 'hits', 'delivery-policy': 'send-by-default'})]),
+            ('Resources', [
+                ('Resource', [
+                    ('Title', parameters['corpname'], {'xml:lang': 'en'}),
+                    ('Description', default_corpus_info['corpus_desc'], {'xml:lang': 'en'}),
+                    ('Languages', [('Language', default_corpus_info['corpus_lang'])]),
+                    ('AvailableDataViews', '', {'ref': 'hits'}),
+                    ('LandingPageUri', self.create_url('first_form', [('corpname', default_corpus_info['corpname'])]))
+                ], {'pid': default_corpus_info['corpname']}),
+            ])
+        ]
+
+        return self._get_element_tree(
+            ns,
+            'EndpointDescription',
+            data, {
+                'xmlns:ed': "http://clarin.eu/fcs/endpoint-description",
+                'version': "2"
+            })
+
+
+    def v2_explain_response(self,
+                            parameters,
+                            explain_response=False):
+        """
+        Creates xml.etree.ElementTree object for fcs explain site.
+        :param parameters: dict
+        :param explain_response: boolean
+        :param ns: str, namespace
+        :return: xml.etree.ElementTree
+        """
+        default_corpus_info = self._corpus_info(parameters['corpname'])
+
+        data = [
+            ('version', parameters['version']),
+            ('record', [
+                ('recordSchema', 'http://explain.z3950.org/dtd/2.0/'),
+                ('recordXMLEscaping', parameters['record_xml_escaping']),
+                ('recordData', [
+                    self._get_element_tree('zr', 'explain', [
+                        ('serverInfo', [
+                            ('host', parameters['server_name']),
+                            ('port', parameters['server_port']),
```

Comment:
Should this be present if the value is `None`?

Reply:
do you mean parameters['server_port']? it's set to 80 by default so it should never be None